Userland/Libraries/LibCore/ConfigFile.cpp

```diff
@@ -36,54 +35,60 @@ ErrorOr<NonnullRefPtr<ConfigFile>> ConfigFile::open_for_system(String const& app
 
 ErrorOr<NonnullRefPtr<ConfigFile>> ConfigFile::open(String const& filename, AllowWriting allow_altering)
 {
-    auto file = File::construct(filename);
-    if (!file->open(allow_altering == AllowWriting::Yes ? OpenMode::ReadWrite : OpenMode::ReadOnly)) {
-        // Failure to open a read-only file is OK, and behaves as if the file was empty.
-        if (allow_altering == AllowWriting::Yes)
-            return Error::from_string_literal("Unable to open config file");
+    auto maybe_file = Stream::File::open(filename, allow_altering == AllowWriting::Yes ? Stream::OpenMode::ReadWrite : Stream::OpenMode::Read);
+    OwnPtr<Stream::BufferedFile> buffered_file;
+    if (maybe_file.is_error()) {
+        // Failure to open a non-existent read-only config file is not an error.
+        if (!(maybe_file.error().code() == ENOENT) && (allow_altering == AllowWriting::No))
```

Comment:
It feels like this could be a lot more clearer if we described the intent directly in the if clause:
```suggestion
        if (!(allow_altering == AllowWriting::Yes && maybe_file.error().code() == ENOENT))
```

Reply:
I removed some parentheses to hopefully make it clearer, and expanded the comment.